### ISSUE TITLE
Pulses symbolic expression error

### ISSUE THREAD

**Issue by wilkensJ** [qiboteam/qibolab#395]
When using symbolic expressions in this way, an error is thrown:

```python
from qibolab.pulses import DrivePulse, Gaussian
from qibolab.symbolic import intSymbolicExpression as se_int
t0 = se_int(0, "t0")
p1 = DrivePulse(t0, 400, 1, 20e6, 0, Gaussian(5), 10)
p1.start = se_int(200, 'tnew')
p1.duration = 200
```
The error is:
```txt
Traceback (most recent call last):
  File "/home/users/jadwiga.wilkens/.conda/envs/testenv/lib/python3.9/site-packages/qibolab/symbolic.py", line 199, in evaluate
    raise Exception()
Exception
```
We tracked down where the error is happening, it is in `symbolic.py` in the `collect_garbage()` method:
https://github.com/qiboteam/qibolab/blob/09b0bcfd4047d31c98312f73f7c60de835f02330/src/qibolab/symbolic.py#L32-L33

In this method the reference count of `tnew` < 4, this is what happens then:
```python
# Before deleting tnew:
# print(SymbolicExpression.instances)
{'t0': t0: 0 = 0, '_p0_start': _p0_start: tnew = 200, '_p0_finish': _p0_finish: (_p0_start + 400) = 600, 'tnew': tnew: 200 = 200}
# After deleting tnew:
# print(SymbolicExpression.instances)
{'t0': t0: 0 = 0, '_p0_start': _p0_start: tnew = InvalidExpressionError, '_p0_finish': _p0_finish: (_p0_start + 400) = InvalidExpressionError}
```
When commenting out the `self.parent.collect_garbage()` line in:
https://github.com/qiboteam/qibolab/blob/09b0bcfd4047d31c98312f73f7c60de835f02330/src/qibolab/symbolic.py#L9-L11
no error is raised anymore.

**Comment by wilkensJ** [comment]
Found out that this works:

```python
from qibolab.pulses import DrivePulse, Gaussian
from qibolab.symbolic import intSymbolicExpression as se_int
t0 = se_int(0, "t0")
p1 = DrivePulse(t0, 400, 1, 20e6, 0, Gaussian(5), 10)
tnew = se_int(200, 'tnew')
p1.start = tnew
# p1.start = se_int(200, 'tnew')
p1.duration = 200
```

But this raises the error from above:

```python
from qibolab.pulses import DrivePulse, Gaussian
from qibolab.symbolic import intSymbolicExpression as se_int
t0 = se_int(0, "t0")
p1 = DrivePulse(t0, 400, 1, 20e6, 0, Gaussian(5), 10)
# tnew = se_int(200, 'tnew')
# p1.start = tnew
p1.start = se_int(200, 'tnew')
p1.duration = 200
```

**Comment by wilkensJ** [comment]
Fixed with PR #350
